sklearn/linear_model/_glm/tests/test_glm.py

```diff
@@ -372,30 +364,21 @@ def test_glm_regression_unpenalized(solver, fit_intercept, glm_dataset):
     # NOT return the minimum norm solution with fit_intercept=True.
     rtol = 5e-5
     if n_samples > n_features:
-        assert model.intercept_ == pytest.approx(intercept)
+        assert model.intercept_ == pytest.approx(intercept, rel=rtol)
         assert_allclose(model.coef_, coef, rtol=rtol)
     else:
-        # As it is an underdetermined problem, prediction = y. The following shows that
-        # we get a solution, i.e. a (non-unique) minimum of the objective function ...
-        assert_allclose(model.predict(X), y, rtol=1e-6)
-        if fit_intercept:
-            # But it is not the minimum norm solution. Otherwise the norms would be
-            # equal.
-            norm_solution = np.linalg.norm(np.r_[intercept, coef])
-            norm_model = np.linalg.norm(np.r_[model.intercept_, model.coef_])
-            assert norm_model > (1 + 1e-12) * norm_solution
-
-            # See https://github.com/scikit-learn/scikit-learn/issues/23670.
-            # Note: Even adding a tiny penalty does not give the minimal norm solution.
-            # XXX: We could have naively expected LBFGS to find the minimal norm
-            # solution by adding a very small penalty. Even that fails for a reason we
-            # do not properly
-        else:
-            # When `fit_intercept=False`, LBFGS naturally converges to the minimum norm
-            # solution on this problem.
-            # XXX: Do we have any theoretical guarantees why this should be the case?
-            assert model.intercept_ == pytest.approx(intercept)
-            assert_allclose(model.coef_, coef, rtol=rtol)
+        # As it is an underdetermined problem, prediction = y. The following
+        # shows that we get a solution, i.e. a (non-unique) minimum of the
+        # objective function ...
+        assert_allclose(model.predict(X), y, rtol=rtol)
+
+        # When initialized from zero weights and intercept, LBFGS naturally
+        # converges to the minimum norm solution on this problem.
+        #
+        # XXX: Do we have any theoretical guarantees why this should be the
+        # case?
```

Comment:
There are papers about gradient descent and minimum norm.

Reply:
Do they cover LBFGS, line search and the combination of stopping criteria we use?

Reply:
Not a direct answer to the question above, but the following quite recent paper seems relevant:

https://opt-ml.org/papers/2020/paper_33.pdf

Maybe @genji would like to comment?

Reply:
I think we should conduct some experiments to see the impact of the intercept init on the test accuracy.

We could also maybe explore how to do the smart intercept init and re-project on the data-span as done in the paper above (not sure if it makes sense for all GLMs).